includes/class-sensei-data-cleaner.php

```diff
@@ -220,8 +228,8 @@ private static function cleanup_options() {
 	private static function cleanup_roles_and_caps() {
 		global $wp_roles;
 
-		// Remove role.
-		remove_role( self::$role );
+		// Ensure that the role exists so we can remove it successfully.
+		add_role( self::$role, self::$role_name );
```

Comment:
I don't quite get why this is being added only to be deleted. As I understand it, `remove_role` will remove the role only if it exists, but wouldn't error if it didn't exist. Could you clarify?

Reply:
This seems similar to what's going on [here](https://github.com/Automattic/sensei/pull/2072#issuecomment-374174620).

Reply:
Yeah, the two are pretty similar, except you're right that in this case we don't get an error if the role doesn't exist. However, if the role doesn't exist then the `remove_role` method [returns early](https://core.trac.wordpress.org/browser/tags/4.9/src/wp-includes/class-wp-roles.php#L180), and doesn't delete all of the data out of the DB. I needed the role to exist in order for the method to delete everything properly.

I don't really like having to do this manually, but it seemed like the simplest solution. A couple possible alternatives:

- Initialize the role through the Sensei code, instead of calling `add_role` directly. We could use the `Sensei_Teacher::create_role` method. I'm not sure if this gives us much benefit over the current solution.

- Switch to direct DB queries or option manipulation to delete the data. I think that this would be a bit fiddly, and I don't like the idea of duplicating the functionality that we get from `remove_role`.

Reply:
My comments from https://github.com/Automattic/sensei/pull/2072#discussion_r175727613 apply here as well. Do what you think is best. 😃

Reply:
Ok, so I looked deeper into this, and it turns out that I was misinterpreting some of what I was seeing. The role needs to be added in order for the tests to work, but I think the problem I was seeing in the code was from messed up data that came from an incomplete implementation earlier. In any case, it's working now without the hack 🎉